StateMachine.cpp

```diff
@@ -53,12 +53,19 @@ void StateMachine::ExternalEvent(BYTE newState, const EventData* pData)
 //----------------------------------------------------------------------------
 void StateMachine::InternalEvent(BYTE newState, const EventData* pData)
 {
-	if (pData == NULL)
+	// if (pData == NULL)
+	// 	pData = new NoEventData();
+#ifndef EXTERNAL_EVENT_NO_HEAP_DATA
+  if (pData == NULL) {
 		pData = new NoEventData();
+	}
+#else
+  assert(pData != NULL);
```

Comment:
I later found we can not generate assert here. If we tried to move state inside a state machine and the state does not require any input data, this will fail. I think if this function is called without data, we should let it.

Reply:
Hmm ok, but I'm not able to link without this patch.
Do you have a different patch?

Reply:
I only removed the assert

```
#ifndef EXTERNAL_EVENT_NO_HEAP_DATA
  if (pData == NULL) {
		pData = new NoEventData();
	}
#endif
```